src/main/kotlin/org/jetbrains/research/ml/dataset/anonymizer/util/FileUtil.kt

```diff
@@ -2,30 +2,54 @@ package org.jetbrains.research.ml.dataset.anonymizer.util
 
 import com.intellij.openapi.application.ApplicationManager
 import com.intellij.openapi.util.io.FileUtil
+import krangl.DataFrame
+import krangl.map
+import krangl.readCSV
+import krangl.writeCSV
+import org.jetbrains.research.ml.dataset.anonymizer.Anonymizer
+import org.jetbrains.research.ml.dataset.anonymizer.util.CodetrackerFileUtil.Language
 import java.io.File
 
 object FileUtil {
     fun createFile(path: String, content: String = ""): File {
         val file = File(path)
         if (!file.exists()) {
-            ApplicationManager.getApplication().runWriteAction {
-                FileUtil.createIfDoesntExist(file)
-                file.writeText(content)
+            ApplicationManager.getApplication().invokeAndWait {
+                ApplicationManager.getApplication().runWriteAction {
+                    FileUtil.createIfDoesntExist(file)
+                    file.writeText(content)
+                }
             }
         }
-     //   AnonymizationTransformation
         return file
     }
 
-//    fun handleLanguageFolder(languageFolderPath: String, outputRootPath: String) {
-//        val languageFolder = File(languageFolderPath)
-//        languageFolder.listFiles()?.filter { it.isDirectory }?.forEach { taskFolder ->
-//            taskFolder.listFiles()?.filter { it.isFile && it.name.endsWith(Extension.Csv.value) }?.forEach {
-//                val df = handleCsvFile(it.path)
-//                val outputPath = "$outputRootPath/${languageFolder.name}/${taskFolder.name}/${it.name}"
-//                val outputFile = createFile(outputPath)
-//                df.writeCSV(outputFile)
-//            }
-//        }
-//    }
+    fun handleLanguageFolder(outputRootPath: String, language: Language, anonymizer: Anonymizer) {
+        val languageFolder = File("$outputRootPath/${language.value}")
```

Comment:
Why it's called 'outputRootPath' when you actually take input files from it?.. You also pass inFile in tests